### PR TITLE
Ft/new hash algo

### PR DESCRIPTION
**Pull request by hugo-ccabral**
Hashes with a new algorithm

Benchmark with a random object (generated by copilot):
image

### CODE REVIEW

File: blocks/loader.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // deno-lint-ignore-file no-explicit-any
-import hash from "https://esm.sh/v135/object-hash@3.0.0";
 import JsonViewer from "../components/JsonViewer.tsx";
 import { ValueType } from "../deps.ts";
+import crypto from "node:crypto";
```
Comment (mcandeia) [inline blocks/loader.ts:4]: you can use crypto without importing from node

Comment (mcandeia) [reply blocks/loader.ts:4]: actually deno has it's own crypto implementation

https://deno.land/api@v1.39.4?s=Crypto

File: benchs/hash.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { webcrypto } from "node:crypto";
```
Comment (mcandeia) [inline benchs/hash.ts:1]: you can create .bench.ts files alongside the file that exports your function that you're going to test. I think this is the standard adopted by deno community

**Review by mcandeia — CHANGES_REQUESTED** [review]

File: blocks/loader.ts (hunk 1 above)
Comment (hugo-ccabral) [reply blocks/loader.ts:4]: but this module is just to create randoms

Comment (hugo-ccabral) [reply blocks/loader.ts]: it is not stable, as my current implementation, so I'm just optimizing the previous approach

File: blocks/loader.bench.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+//     for (const test of tests) {
+//         await md5(JSON.stringify(test));
+//     }
+// });
+
+// Deno.bench("hash-wasm sha1", { group: "object-hash" }, async () => {
+//     for (const test of tests) {
+//         await sha1(JSON.stringify(test));
+//     }
+// });
```
Comment (matheusgr) [inline blocks/loader.bench.ts:93]:
```suggestion
```

**Review by matheusgr — APPROVED** [review]

File: blocks/loader.ts [hunk 4]
```diff
@@ -192,11 +192,16 @@ const wrapLoader = ({
...
+              ignoreUnknown: true,
+              respectType: false,
+              respectFunctionProperties: false,
+              algorithm: "md5",
+            });
+            span.setAttributes({
+              hash_size_bytes: hashedProps.length * 2,
+              props_length: JSON.stringify(props).length,
+            });
             url.searchParams.set("props", hashedProps);
```
Comment (matheusgr) [inline blocks/loader.ts:205]: could this not run if otel is not active? if so, this is a bug. it should always run set("props").

**Review by matheusgr — APPROVED** [review]

File: engine/schema/schemeable.test.ts [hunk 5]
```diff
@@ -77,7 +77,7 @@ Deno.test("Simple type generation", async () => {
     fail("SimpleType should exists");
   }
 
-  const name = Deno.build.os === "windows" ? "tl@157-178" : "tl@155-174";
+  const name = "tl@155-174";
```
Comment (mcandeia) [inline engine/schema/schemeable.test.ts:80]: Will this pass even in different OSs?

**Review by mcandeia — APPROVED** [review]

File: engine/schema/schemeable.test.ts (hunk 5 above)
Comment (hugo-ccabral) [reply engine/schema/schemeable.test.ts:80]: I'm fixing now so don't break the build. But my check is broken locally, even with my Deno.build.os returning "windows"

File: blocks/loader.ts (hunk 4 above)
Comment (hugo-ccabral) [reply blocks/loader.ts:205]: no, it will always run. Even if otel disabled

**Pull request merged**